cmake/sdk-support.cmake

```diff
@@ -0,0 +1,128 @@
+  endif()
+  set(${var} ${install_path} PARENT_SCOPE)
+endfunction()
+
+function(llvm_mos_sdk_add_include_files)
+  get_relative_install_path(inc_path)
+  install(FILES ${ARGN} DESTINATION ${inc_path})
+endfunction()
+
+function(llvm_mos_sdk_add_linker_scripts)
+  get_relative_install_path(script_path)
+  install(FILES ${ARGN} DESTINATION ${script_path})
+endfunction()
+
+function(llvm_mos_sdk_install_targets)
+  get_relative_install_path(lib_path)
+  install(TARGETS ${ARGN} DESTINATION ${lib_path})
+endfunction()
+
+function(llvm_mos_sdk_track_platform_target name)
+  set_property(GLOBAL APPEND PROPERTY LLVM_MOS_CURRENT_PLATFORM_TARGETS ${name})
+endfunction()
+
+function(llvm_mos_sdk_add_crt)
+  set(target_name ${LLVM_MOS_CURRENT_PLATFORM}_crt)
+  add_library(${target_name} STATIC $<TARGET_PROPERTY:common_crt,INTERFACE_SOURCES> ${ARGN})
+  set_target_properties(${target_name} PROPERTIES OUTPUT_NAME crt)
+  llvm_mos_sdk_install_targets(${target_name})
+  llvm_mos_sdk_track_platform_target(${target_name})
+endfunction()
+
+function(llvm_mos_sdk_add_c)
+  set(target_name ${LLVM_MOS_CURRENT_PLATFORM}_c)
+  add_library(${target_name} STATIC $<TARGET_PROPERTY:common_c,INTERFACE_SOURCES> ${ARGN})
+  set_target_properties(${target_name} PROPERTIES OUTPUT_NAME c)
+  llvm_mos_sdk_install_targets(${target_name})
+  llvm_mos_sdk_track_platform_target(${target_name})
+endfunction()
+
+function(llvm_mos_sdk_add_object_file name src)
+  set(target_name ${LLVM_MOS_CURRENT_PLATFORM}_${name})
+
+  if (ARGC GREATER 2)
+    set(outname ${ARGV2})
+  else()
+    set(outname ${target_name})
+  endif()
+
+  # Even though we just need the object file, build a full static library
+  # so POST_BUILD hook works (just ignore the .a, it won't be installed).
+  add_library(${target_name} STATIC ${src})
+  get_relative_install_path(lib_path)
+  install(FILES $<TARGET_OBJECTS:${target_name}> DESTINATION ${lib_path} RENAME ${outname})
+  llvm_mos_sdk_track_platform_target(${target_name})
+
+  # Create symlink (or copy) to the object file on each build.
+  add_custom_command(TARGET ${target_name} POST_BUILD COMMAND ${CMAKE_COMMAND} ARGS -DNAME=${outname}
+          -DTARGET=$<TARGET_OBJECTS:${target_name}> -DOUTDIR=${CMAKE_CURRENT_BINARY_DIR}
+          -P ${CMAKE_SOURCE_DIR}/cmake/create-symlink.cmake DEPENDS $<TARGET_OBJECTS:${target_name}>)
+endfunction()
+
+function(llvm_mos_sdk_install_symlink name dest)
+  set(full_name ${name}${CMAKE_EXECUTABLE_SUFFIX})
+  set(full_dest ${dest}${CMAKE_EXECUTABLE_SUFFIX})
+
+  install(SCRIPT ${CMAKE_SOURCE_DIR}/cmake/install-symlink.cmake
+          CODE "install_symlink(${full_name} ${full_dest} bin)")
+endfunction()
+
+function(llvm_mos_sdk_add_platform name)
+  cmake_parse_arguments(ARG BUILD_EXAMPLES "" "" ${ARGN})
+  set(cfg_file mos-${name}.cfg)
+
+  if(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${cfg_file})
+    message(FATAL_ERROR "${cfg_file} does not exist for llvm_mos_sdk_add_platform.")
+  endif()
+
+  if(NOT MOS)
```

Comment:
Can you provide a bit of commentary here about the intended purpose of both the MOS and non-MOS versions of this function? It's a bit surprising that this routine can be used in either context, especially since little code appears to be shared between the two flows.

Reply:
Added as a comment on `llvm_mos_sdk_add_platform`